PWGHF/HFC/Tasks/taskCorrelationD0.cxx

```diff
@@ -0,0 +1,501 @@
+    registry.fill(HIST("hNtracksMixing"), Ntracks);
+  }
+
+  template <typename TTracks>
+  void fillMFTQA(float multiplicity, TTracks tracks)
+  {
+    for (auto& track1 : tracks) {
+      registry.fill(HIST("hEtaMFT"), track1.eta());
+      float phi = track1.phi();
+      o2::math_utils::bringTo02Pi(phi);
+      registry.fill(HIST("hPhiMFT"), phi);
+      registry.fill(HIST("hEtaPhiMFT"), multiplicity, track1.eta(), phi);
+    }
+  }
+
+  //  TODO: Check how to put this into a Filter
+  template <typename TTrack>
+  bool isAcceptedCandidate(TTrack candidate)
+  {
+    if (!(candidate.hfflag() & 1 << DecayType::D0ToPiK)) {
+      return false;
+    }
+    if (cutYCandMax >= 0. && std::abs(YD0(candidate)) > cutYCandMax) {
+      return false;
+    }
+    return true;
+  }
+
+  //  TODO: Note: we do not need all these plots since they are in D0 and Lc task -> remove it after we are sure this works
+  template <typename TTracks>
+  void fillCandidateQA(TTracks candidates)
+  {
+    for (auto& candidate : candidates) {
+      if (!isAcceptedCandidate(candidate)) {
+        continue;
+      }
+
+      if (candidate.isSelD0() >= d_selectionFlagD0) {
+        registry.fill(HIST("hMass"), InvMassD0(candidate), candidate.pt());
+      }
+      if (candidate.isSelD0bar() >= d_selectionFlagD0bar) {
+        registry.fill(HIST("hMass"), InvMassD0bar(candidate), candidate.pt());
+      }
+
+      registry.fill(HIST("hPtCand"), candidate.pt());
+      registry.fill(HIST("hPtProng0"), candidate.ptProng0());
+      registry.fill(HIST("hPtProng1"), candidate.ptProng1());
+      registry.fill(HIST("hDecLength"), candidate.decayLength(), candidate.pt());
+      registry.fill(HIST("hDecLengthXY"), candidate.decayLengthXY(), candidate.pt());
+      registry.fill(HIST("hd0Prong0"), candidate.impactParameter0(), candidate.pt());
+      registry.fill(HIST("hd0Prong1"), candidate.impactParameter1(), candidate.pt());
+      registry.fill(HIST("hd0d0"), candidate.impactParameterProduct(), candidate.pt());
+      registry.fill(HIST("hCTS"), CosThetaStarD0(candidate), candidate.pt());
+      registry.fill(HIST("hCt"), CtD0(candidate), candidate.pt());
+      registry.fill(HIST("hCPA"), candidate.cpa(), candidate.pt());
+      registry.fill(HIST("hEtaCand"), candidate.eta(), candidate.pt());
+      registry.fill(HIST("hSelectionStatus"), candidate.isSelD0() + (candidate.isSelD0bar() * 2), candidate.pt());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter0(), candidate.pt());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter1(), candidate.pt());
+      registry.fill(HIST("hDecLenErr"), candidate.errorDecayLength(), candidate.pt());
+      registry.fill(HIST("hDecLenXYErr"), candidate.errorDecayLengthXY(), candidate.pt());
+    }
+  }
+
+  template <typename TTarget, typename TTracksTrig, typename TTracksAssoc>
+  void fillCorrelations(TTarget target, TTracksTrig tracks1, TTracksAssoc tracks2, float multiplicity, float posZ)
```

Comment:
It could be worth to add some upfront selection on the eta, pT, etc. of trigger and associated particles, to reduce the size of the containers, but this is your choice.